### DISCUSSION TITLE
Check for StrEnum initialization values

### DISCUSSION THREAD

**Discussion by Jonas1312** [microsoft/pyright#7322] [Ideas]
Hi,

I have an issue with StrEnum. pyright does not find any error in this simple code:

```python
from enum import StrEnum


class Color(StrEnum):
    red = "red"
    green = "green"
    blue = "blue"


class Fruit(StrEnum):
    apple = "apple"
    banana = "banana"
    cherry = "cherry"


a = Fruit(Color.red)
b = Fruit(Color.red.value)

```

This is because Fruit init method is typed like this: `class Fruit(value: str)`

So I wonder if pyright could/should catch those errors?

If not, what are good alternatives to python enums for catching those errors before runtime?

Thanks!

**Comment by erictraut** [comment]
I'm not aware of any tools that would detect this statically. From a type analysis standpoint, there is no type violation here, so it shouldn't be surprising that neither pyright nor mypy generate an error here. Perhaps there are linter rules that would help you catch such errors?
